.ci/Jenkinsfile

```diff
@@ -124,9 +124,16 @@ pipeline {
       post {
         always {
           archiveArtifacts(allowEmptyArchive: true, artifacts: '${BASE_DIR}/**/mail-out-test*.html')
-          junit(allowEmptyResults: true,
-            keepLongStdio: true,
-            testResults: "${BASE_DIR}/target/surefire-reports/junit-*.xml,${BASE_DIR}/target/surefire-reports/TEST-*.xml")
+          withEnv([
```

Comment:
to report to Otel, Do you have to pass those env vars? if so, it means that you have to edit the Jenkinsfile on those projects you want to report Otel data from a JUnit file, in that case, I do not see the advantage of overwriting the unit step.

Reply:
We could infer those values using a company/team wide convention:

- OTEL_SERVICE_NAME == env.REPO
- JUNIT_OTEL_SERVICE_VERSION (if isPR then "use PR$ID", else if isBranch then "use branch name" else if isTag "use tag name") else "use undefined"?
- JUNIT_OTEL_TRACE_NAME == "env.REPO-junit-tests"

Then we would not need to set the env in the outer pipeline

Reply:
@kuisathaverat @v1v any suggestion to unblock this?

Reply:
What do you think if we split this implementation in two stages:
1) To support the OTEL junit tracing with a new shared library step. Then we test in a couple of place and see how it goes.
2) To discuss if the `junit` can be the one to use broadly this junit implementation and then apply the changes.

T

Reply:
I do not like it, but I am not blocking it

Reply:
And about the predefined name convention?

I'd see it with good eyes to have another step: `junitWithOtel` and test it elsewhere

Reply:
@kuisathaverat @v1v I've updated the code renaming the step to `junit2Otel` so that it does not override the default junit step

Reply:
If you agree, we can merge this one

Reply:
Changes needed in the `.ci/Jenkinsfile` to use `junit2Otel` ?

Reply:
Once released we can include it in the current Jenkinsfile. With latest changes, becuase the step will evaluate the env.REPO for the service/trace name, and isPR/isTag/isBranch to get the current version, the change will be simply changing the `junit` call for `junit2Otel` (⚠️ camelcase)